src/ray/gcs/client.cc

```diff
@@ -54,9 +145,15 @@ Status Attach(plasma::EventLoop &event_loop) {
 }
 
 Status AsyncGcsClient::Attach(boost::asio::io_service &io_service) {
-  asio_async_client_.reset(new RedisAsioClient(io_service, context_->async_context()));
-  asio_subscribe_client_.reset(
-      new RedisAsioClient(io_service, context_->subscribe_context()));
+  // Take care of sharding contexts.
+  shard_asio_async_clients_.clear();
+  shard_asio_subscribe_clients_.clear();
```

Comment:
Why are the `clear()` calls necessary? Do we expect them to have something in them?

I would lean towards making sure `Attach` is only called once, and I would change this to a `RAY_CHECK(shard_asio_async_clients_.empty())`.

Reply:
Sounds good :)